### ISSUE TITLE
Create a data structure for each listable menu items and add them to the nav

### ISSUE THREAD

**Issue by praveenscience** [CatsInTech/Rezume#8]
For the following elements, there's no proper data structure defined:

- Experience
- Education
- Awards
- Projects
- Skills
- Hobbies

Can we have a data structure to store each of them as lists? I guess most of them (except Skills & Hobbies) need to be sorted by dates. We might also need a good data structure! 😊 Let's discuss and this is again up for grabs - there won't be any file changes, but it's going to be associated with other issues - individually discussing the structures. This is again up for grabs! 😊

**Comment by anushreejain98** [comment]
@praveenscience I am interested to work on it. Discussion of some ideas: https://gssoc20.slack.com/archives/C0109EBE74G/p1585217280017900?thread_ts=1585164117.013600&cid=C0109EBE74G

**Comment by anushreejain98** [comment]
For Experience consider following fields:

Select from Internship- industrial or research /FTE
Company/Institute
Duration
Project Title

For Projects:
Project Title
Techstack/skills used
Brief Description

Experience can be linked with projects through title

**Comment by anushreejain98** [comment]
For Skills

Programming languages - (c++, python....) 
Web Technologies- (HTML, CSS etc..) 
Database management: (mySQL, mongodb, Cassandra, redis etc.) 
Data Analytics / python ML libraries- (numpy, pandas, scikitlearn etc. ) 
Operating System: (linux, windows etc.) 
Others/miscellaneous:-(latex, git etc.)

**Comment by anushreejain98** [comment]
Education:
Degree/qualification: (btech, matriculation,secondary) 
Passing Year:
Institution/Board: 
CPI/Percentage

**Comment by praveenscience** [comment]
Wow, that looks awesome @anushreejain98. Keep up the good work! 😊

**Comment by imabp** [comment]
Sir, for #16  I would like to use mapping technique to store the skills of the user. Below is the image representation.
<b>We have two schemas, having details as per @anushreejain98  said, and the second schema which will help us to find how many developers we have with that skill. Therefore it will be helpful for quick analysis of the organisation. If a new skill is entered, by user, it will be stored in the mapped schema, with count as 1. Any existing skill, will be just incremented respectively in the second schema.</b>
![image](https://user-images.githubusercontent.com/53480076/77645334-94bef780-6f88-11ea-915e-4e66f153b158.png)

**Comment by praveenscience** [comment]
@imabp Yea, that's a good idea. But I guess we need to do this on the server side and resolve the names alone on the client side. That works right? 😊

**Comment by imabp** [comment]
@praveenscience  yes sir,  I see, we can use mongoDB for this stuff to handle out.

**Comment by praveenscience** [comment]
@imabp We are using MySQL on the production machine - we can use Knex to connect, if needed.

**Comment by imabp** [comment]
Wow @praveenscience  I didnt know about this. Thanks !!!! so much

**Comment by anushreejain98** [comment]
> Wow, that looks awesome @anushreejain98. Keep up the good work!

Thanks @praveenscience , could you help us with a sample or reference of how to implement it?

**Comment by praveenscience** [comment]
@anushreejain98 Yes, soon. Give me some time. I am in a call - might take an hour max.

**Comment by anushreejain98** [comment]
@praveenscience could give us a go ahead on how to start with implementation of this issue?

**Comment by praveenscience** [comment]
@anushreejain98 Did you miss the last session, where I was talking about React? 😔

**Comment by praveenscience** [comment]
Team, let's decide on how to take this forward... 😊 I am ready to discuss about this anytime... 😊

**Comment by praveenscience** [comment]
We'll discuss the data structure tomorrow morning BST / afternoon IST. Whoever is contributing to the above issues, please make sure you attend this. 😊

**Comment by praveenscience** [comment]
Individual Label:

```javascript
{
  Type: "Text",
  Label: "Full Name",
  Options: [{
    Name: "",
    Value: ""
  }]
}
```

The whole thing we discussed, and it's WIP:

```javascript
const About = {
  Name: {},
  Location: {},
  CurrentPosition: {},
  Summary: {}
};

const Contact = {
  Email: {},
  Website: {},
  Phone: {}
};

const Experience = {
  ExpTitle: {},
  ExpType: {},
  Employer: {},
  EmployerURL: {},
  ExpStartDate: {},
  ExpEndDate: {},
  ExpCurrent: {},
  ExpLocation: {},
  ExpDesc: {}
};

const Education = {
};

const Awards = {
};

const Projects = {
  ProjTitle: {},
  TechStack: {},
  ProjDesc: {},
  ProjLocation: {},
  ProjEmp: {},
  ProjURL: {}
};

const Skills = {
  SkillName: {},
  SkillExp: {},
  SkillCategory: {}
};

const Hobbies = {
};

export default {
  ...About,
  ...Experience,
  ...Education,
  ...Awards,
  ...Projects,
  ...Skills,
  ...Hobbies,
  ...Contact
};
```

**Comment by csharp489** [comment]
@praveenscience Please check this format for education. 
Add Experience
	1. Graduation
	2.Post Graduation
	3.Diploma
	4.PHD
Graduation Status
	a.1.Pursuing
	a.2.Completed
	b College
	c Start Year (Drop down)
	d End Year(Drop down)
	e Degree
	f Stream
	g Performance( based on GPA)

**Comment by adad20** [comment]
@praveenscience for experience we can consider the following:
`experience:[{
        title:{
            type: String,
            required: true
        },
        company:{
            type: String,
            required: true
        },
        location:{
            type: String
        },
        from:{
            type: Date,
            required: true
        },
        to:{
            type: Date,
            required: true
        },
        current:{
            type: Boolean,
            default: false
        },
        description:{
            type: String
        }
    }]`

**Comment by praveenscience** [comment]
@anushreejain98 Just the only thing is, the key names need to be unique. Like title, from, to, current, description, location - all these can be shared by education as well.

**Comment by lakshyajit165** [comment]
@praveenscience, I think the hobbies could have 2 fields - name and type, for example:
 - name - Singing, type - Music
 - name - Cricket, type - Sports

**Comment by Shivansh-yadav13** [comment]
@praveenscience, we can use badges ([shields.io](https://shields.io/)) for skills, what do you think?

**Comment by praveenscience** [comment]
@Shivansh-yadav13 That's a great idea.

**Comment by himanshusanecha** [comment]
Array data structure to store all projects or experiences or heavy text data can exceed the mongodb limit (if I am not wrong), like projects can contain images, experiences can contain some work samples etc. etc. 

Also updating individual experience or project would require aggregation I guess.

**Comment by praveenscience** [comment]
@himanshusanecha We never use MongoDB, didn't check the tech stack for this project? Why using MongoDB BTW? We're working on a pure Front End application here.

Project containing images on Resume? I guess we can start with something simpler man.

**Comment by praveenscience** [comment]
Okay, tell me more about the aggregation @himanshusanecha. I didn't quite understand or get it.

**Comment by himanshusanecha** [comment]
> @himanshusanecha We never use MongoDB, didn't check the tech stack for this project? Why using MongoDB BTW? We're working on a pure Front End application here.
> 
> Project containing images on Resume? I guess we can start with something simpler man.

Oh sorry, I thought it is a full fledged application to maintain database for resume's.
Ignore aggregation it is related to querying from the database.
